Build fails on Debian 11 (glibc 2.31)

I recently wanted to try openmohaa on my ppc64el workstation that runs Debian 11.11.

Trying to compile it from sources (8cc1cd6), the build failed trying to link:

```
[100%] Linking CXX executable openmohaa.ppc64el
/usr/bin/ld: CMakeFiles/openmohaa.dir/code/client/libmumblelink.c.o: référence au symbole non défini « shm_open@@GLIBC_2.17 »
/usr/bin/ld : /lib/powerpc64le-linux-gnu/librt.so.1 : erreur lors de l'ajout de symboles : DSO manquant dans la ligne de commande
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/openmohaa.dir/build.make:2201 : openmohaa.ppc64el] Erreur 1
make[1]: *** [CMakeFiles/Makefile2:406 : CMakeFiles/openmohaa.dir/all] Erreur 2
make: *** [Makefile:149 : all] Erreur 2
```
I had to add `-lrt` in `./CMakeFiles/openmohaa.dir/link.txt` and that linked successfully. I was able to start the game!

This option is no longer required to use shm_open beginning with glibc 2.34. But it seems Debian 11 has glibc 2.31.

So maybe this option could be added, at least depending on the glibc version.

Note: Trying to run executables provided in v0.81.0, they don't start because the minimum glibc version is not reached:
```
$ ./openmohaa.ppc64el
./openmohaa.ppc64el: /lib/powerpc64le-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by ./openmohaa.ppc64el)
./openmohaa.ppc64el: /lib/powerpc64le-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by ./openmohaa.ppc64el)
```

--- a/code/sys/CMakeLists.txt
+++ b/code/sys/CMakeLists.txt
@@ -62,6 +62,11 @@ elseif(UNIX)
 		endif()
 	endif()
 	target_link_libraries(syslib INTERFACE stdc++)
+
+	if (NOT APPLE)
+		# For when using GLIBC versions older than 2.34
+		target_link_libraries(syslib INTERFACE rt)
+	endif()
 endif()
 
 if (APPLE)

